operator/v1/types_network.go

```diff
@@ -332,6 +336,14 @@ type HybridOverlayConfig struct {
 	HybridOverlayVXLANPort *uint32 `json:"hybridOverlayVXLANPort,omitempty"`
 }
 
+type IPsecConfig struct {
+	// enableIPsec enables IPsec encryption for pod-to-pod traffic on the pod network
+	// within the cluster. Default is false.
+	// +optional
+	// +kubebuilder:default:=false
+	EnableIPsec bool `json:"enableIPsec"`
```

Comment:
not sure if this should be just `Enable`...

    ipsecConfig:
      enable: true

seems clear enough, while

    ipsecConfig:
      enableIPsec: true

is a little redundant. @sttts may have opinions

otherwise
/lgtm

Reply:
I think  just "enable" looks clearer TBH. I will change it.

Reply:
Please re-add LGTM is you are happy